traffic_ops/v3-client/profile.go

```diff
@@ -126,16 +126,16 @@ func (to *Session) GetProfileByName(name string) ([]tc.Profile, toclientlib.ReqI
 	return to.GetProfileByNameWithHdr(name, nil)
 }
 
-func (to *Session) GetProfileByParameterWithHdr(param string, header http.Header) ([]tc.Profile, toclientlib.ReqInf, error) {
-	URI := fmt.Sprintf("%s?param=%s", APIProfiles, url.QueryEscape(param))
+func (to *Session) GetProfileByParameterWithHdr(param int, header http.Header) ([]tc.Profile, toclientlib.ReqInf, error) {
```

Comment:
You can't change the signatures of these methods, if this code is not a part of a major release. There might be other client scripts using these methods, which will break if you change the signature. Instead, I would suggest doing something like you did in the `v2` client method above. That way, the method signature stays the same.